lib/manageiq/messaging/kafka/client.rb

```diff
@@ -0,0 +1,62 @@
+module ManageIQ
+  module Messaging
+    module Kafka
+      class Client < ManageIQ::Messaging::Client
+        require 'kafka'
+        require 'manageiq/messaging/kafka/common'
+        require 'manageiq/messaging/kafka/queue'
+        require 'manageiq/messaging/kafka/background_job'
+        require 'manageiq/messaging/kafka/topic'
+
+        include Common
+        include Queue
+        include BackgroundJob
+        include Topic
+
+        private *delegate(:subscribe, :unsubscribe, :publish, :to => :kafka_client)
+        delegate :close, :to => :kafka_client
+
+        attr_accessor :encoding
+
+        private
+
+        attr_reader :kafka_client
+
+        # @options options :host
+        # @options options :hosts (array)
+        # @options options :port
+        # @options options :group_ref
+        # @options options :client_ref (optional)
+        # @options options :encoding (default to 'yaml')
+        def initialize(options)
+          hosts = Array(options[:hosts]) + Array(options[:host])
+          hosts.collect! { |host| "#{host}:#{options[:port]}" }
+
+          @encoding = options[:encoding] || 'yaml'
+          require "json" if @encoding == "json"
+
+          connection_opts = {}
+          connection_opts[:client_id] = options[:client_ref] if options[:client_ref]
+
+          @kafka_client = ::Kafka.new(hosts, connection_opts)
+        end
+
+        public
```

Comment:
could we put the public methods up top and move the private methods down lower in the file?